src/tribler-core/tribler_core/session.py

```diff
@@ -424,6 +424,7 @@ async def start(self):
             ipv8_config['address'] = self.config.get_ipv8_address()
             ipv8_config['overlays'] = []
             ipv8_config['keys'] = []  # We load the keys ourselves
+            ipv8_config['working_directory'] = str(self.config.get_state_dir() / 'sqlite')
```

Comment:
Isn't the `sqlite` directory appended by the IPv8 database logic?

Reply:
Good point, is the `self.config.get_state_dir()` safe to write to for IPv8-based overlays or should this go into `self.config.get_state_dir() / 'ipv8'`?

Reply:
`self.config.get_state_dir()` should be safe to write to. Please test it with Tribler to see if it ends up in the right directory (e.g., by reverting the lazy loading commit).

Reply:
@devos50 I made that change and verified:

 - No `identity.db` or `attestations.db` is created on a normal run.
 - When lazy loaded in, `identity.db` and `attestations.db` are created in `.Tribler/7.5/sqlite`

Reply:
Assuming that is expected behavior, I'll mark this as READY.